src/electron/process_manager.ts

```diff
@@ -205,33 +177,19 @@ export class ConnectionManager {
     console.log('restarting tun2socks after resume');
 
     this.tun2socks.onExit = this.tun2socksExitListener;
-    this.tun2socks.start(this.isUdpEnabled);
+    this.tun2socks.start(!this.isAutoConnect);
```

Comment:
This is scary: this code kicks off an async start of tun2socks, and then attempts to restart it without waiting for it to start.  It seems hard to be sure that this doesn't have any race conditions.  Could this call use `await` instead?

Reply:
This was a mistake on my part. We should only (re)start tun2socks once and leverage the UDP check. The routing service sends a 'reconnected' event on resume, after the network is re-established. To avoid a race condition with the network change listener, this method now only restores tun2socks' exit listener.